torch/distributed/_functional_collectives.py

```diff
@@ -141,15 +158,23 @@ def __new__(cls, elem: torch.Tensor):
             device=elem.device, requires_grad=False
         )
         r.elem = elem
+        r._holder = holder
         return r
 
     def __repr__(self):
         return f"AsyncCollectiveTensor({self.elem})"
 
+    def trigger_wait(self):
+        self._holder.wait_tensor()
+        return self
+
     @classmethod
     def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
-        def unwrap(e: Any):
-            return wait_tensor(e.elem)
+        def unwrap(e: AsyncCollectiveTensor):
+            # TODO do we need to insert a wait_tensor op for all tensors or just the first one?
+            # Given only first usage of any of the output tensors needs to wait, for now we emit only one wait
```

Comment:
Is this the reason why`WaitHolder` is introduced? This is a contract to any graph transformation involving collective. So it may be good to well-document this design.

Reply:
The follow up PR fixes this to relax this contract and make it more intuitive. All result tensors are waitable.